### PR TITLE
PIM-6118: Avoid BC breaks in minor release

### PR DESCRIPTION
**Pull request by a2xchip**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

PIM-6118 introduced some BC breaks. This PR is to avoid BC breaks in 1.7 and use `soft deprecation` instead

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | OK
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Form/Type/AttributeRequirementType.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Attribute requirement form type
+ *
+ * @deprecated No longer used
+ * @todo       To be removed in upcoming 1.8 release (@a2xchip)
+ * @author     Gildas Quemener <gildas@akeneo.com>
+ * @copyright  2013 Akeneo SAS (http://www.akeneo.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Form/Type/AttributeRequirementType.php]: Please respect the other "@deprecated" classes, with the same semantic "will be removed ... etc"

File: src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/SetAttributeRequirements.php [hunk 2]
```diff
@@ -21,16 +23,36 @@ class SetAttributeRequirements extends AbstractMassEditOperation
     protected $channelRepository;
 
     /**
+     * @var AttributeRepositoryInterface
+     * @deprecated Not used anymore
+     * @todo       To be removed in upcoming 1.8 release (@a2xchip)
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/MassEditAction/Operation/SetAttributeRequirements.php]: same

File: src/Pim/Bundle/EnrichBundle/Resources/config/form_types.yml [hunk 3]
```diff
@@ -95,6 +96,13 @@ services:
         tags:
             - { name: form.type, alias: pim_enrich_family }
 
+    pim_enrich.form.type.attribute_requirement:
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/config/form_types.yml]: Add @deprecated here too

**Comment by a2xchip** [comment]
:tada: :tada: :tada:

**Pull request merged**